File: frontend/src/views/components/toolbar-layout.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@polymer/iron-icon/iron-icon.js';
import '@vaadin/button/src/vaadin-button.js';
import '@vaadin/select/src/vaadin-select.js';

@customElement('toolbar-layout')
export class ToolbarLayout extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-horizontal-layout class="content" style="width: 100%;">
 <vaadin-button theme="icon tertiary" aria-label="Add new" id="addBtn" tabindex="0">
  <iron-icon icon="lumo:plus"></iron-icon>
 </vaadin-button>
 <vaadin-button theme="icon error tertiary" aria-label="Add new" id="deleteBtn" tabindex="0">
  <iron-icon icon="lumo:cross"></iron-icon>
 </vaadin-button>
 <vaadin-button theme="icon tertiary" aria-label="Add new" id="saveBtn" tabindex="0">
  <iron-icon icon="lumo:upload"></iron-icon>
 </vaadin-button>
 <vaadin-button theme="icon tertiary success" aria-label="Add new" id="downloadBtn" tabindex="0">
  <iron-icon icon="lumo:download"></iron-icon>
 </vaadin-button>
 <vaadin-button theme="icon tertiary" aria-label="Add new" id="reloadBtn" tabindex="0">
  <iron-icon icon="lumo:reload"></iron-icon>
 </vaadin-button>
 <div style="margin-left: auto;">
  <vaadin-button theme="icon tertiary" aria-label="Add new" id="previousBtn" tabindex="0">
   <iron-icon icon="lumo:angle-left"></iron-icon>
  </vaadin-button>
  <span id="pagingSummary">50/120</span>
  <vaadin-button theme="icon tertiary" aria-label="Add new" tabindex="0" id="nextBtn">
   <iron-icon icon="lumo:angle-right"></iron-icon>
  </vaadin-button>
  <vaadin-select value="Item one" id="pagingSelect" style="width: 75px;" theme="small" placeholder="10"></vaadin-select>
 </div>
</vaadin-horizontal-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: src/main/java/sr/we/ui/components/ToolbarLayout.java
package sr.we.ui.components;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.select.Select;
import com.vaadin.flow.component.template.Id;
import sr.we.ui.views.LineAwesomeIcon;

/**
 * A Designer generated component for the toolbar-layout template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("toolbar-layout")
@JsModule("./src/views/components/toolbar-layout.ts")
public class ToolbarLayout extends LitTemplate {

    @Id("addBtn")
    private Button addBtn;
    @Id("deleteBtn")
    private Button deleteBtn;
    @Id("saveBtn")
    private Button saveBtn;
    @Id("downloadBtn")
    private Button downloadBtn;
    @Id("reloadBtn")
    private Button reloadBtn;
    @Id("previousBtn")
    private Button previousBtn;
    @Id("pagingSummary")
    private Span pagingSummary;
    @Id("nextBtn")
    private Button nextBtn;
    @Id("pagingSelect")
    private Select pagingSelect;

    /**
     * Creates a new ToolbarLayout.
     */
    public ToolbarLayout() {
        addBtn.getElement().removeAllChildren();
        deleteBtn.getElement().removeAllChildren();
        saveBtn.getElement().removeAllChildren();
        downloadBtn.getElement().removeAllChildren();
        reloadBtn.getElement().removeAllChildren();
        previousBtn.getElement().removeAllChildren();
        nextBtn.getElement().removeAllChildren();
        // You can initialise any data required for the connected UI components here.
        addBtn.setIcon(new LineAwesomeIcon("la la-plus"));
        deleteBtn.setIcon(new LineAwesomeIcon("la la-trash"));
        saveBtn.setIcon(new LineAwesomeIcon("la la-save"));
        downloadBtn.setIcon(new LineAwesomeIcon("la la-download"));
        reloadBtn.setIcon(new LineAwesomeIcon("la la-redo-alt"));
        previousBtn.setIcon(new LineAwesomeIcon("la la-angle-left"));
        nextBtn.setIcon(new LineAwesomeIcon("la la-angle-right"));
    }

}
